src/components/AddToCartButton.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from "react";
+import Button from "react-bootstrap/Button";
+import { createOrder, addProductToOrder, getProductById } from "./api/index";
+
+import { getLocalCart, storeLocalCart } from "./auth/auth";
+
+const AddToCartButton = ({
+  quant,
+  setAddedToCart,
+  productId,
+  price,
+  currentUser,
+  setCurrentOrder,
+  currentOrder,
+}) => {
+  const handleAddToCart = async (event) => {
+    event.preventDefault();
+    const id = productId;
+    const quantity = quant;
+    try {
+      setAddedToCart(true);
```

Comment:
one way of solving this is to move the logic to the parent component.

but here, at least move setAddedToCart to after we know it worked! at the end of the try {}